src/lib/utils/synapseTypes/EntityHeader.ts

```diff
@@ -1,20 +1,38 @@
 import { ConcreteEntityType } from './ConcreteEntityType'
 
-// https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityHeader.html
+/**
+ * [EntityHeader | Synapse REST Docs](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityHeader.html)
+ */
 export type EntityHeader = {
-  name: string //	The name of the entity
-  id: string //	The id of the entity
-  type: ConcreteEntityType //	The type of the entity
-  versionNumber: number //	The version number of the entity
-  versionLabel: string //	The user defined version label of the entity
-  benefactorId: number //	The ID of the entity that this Entity's ACL is inherited from.
-  createdOn: string //	The date this entity was created.
-  modifiedOn: string //	The date this entity was last modified.
-  createdBy: string //	The ID of the user that created this entity.
-  modifiedBy: string //	The ID of the user that last modified this entity.
+  /**	The name of the entity */
+  name: string
+  /**	The id of the entity */
+  id: string
+  /**	The type of the entity */
+  type: ConcreteEntityType
+  /** The version number of the entity */
+  versionNumber: number
+  /**	The user defined version label of the entity */
+  versionLabel: string
+  /**	The ID of the entity that this Entity's ACL is inherited from. */
+  benefactorId: number
+  /**	The date this entity was created. */
+  createdOn: string
+  /**	The date this entity was last modified. */
+  modifiedOn: string
+  /**	The ID of the user that created this entity. */
+  createdBy: string
+  /**	The ID of the user that last modified this entity. */
+  modifiedBy: string
 }
 
-// https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityPath.html
+/**
+ * The full path for an entity. Note that the root entity, syn4489, may be included.
+ * The docs do not indicate that most fields are omitted from the EntityHeader objects, but our custom type
+ * signature captures this.
+ *
+ * [EntityPath | Synapse REST Docs](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityPath.html)
+ */
 export type EntityPath = {
-  path: EntityHeader[] // The list of all entities in this entity's path. The first element is the root parent and the last element (n) is the entity.
+  path: Pick<EntityHeader, 'name' | 'id' | 'type'>[] // The list of all entities in this entity's path. The first element is the root parent and the last element (n) is the entity.
 }
```

Comment:
I caught this nuance in the entity path service while creating some of the mocks, so any `EntityHeader`-related changes in this PR were caused by this.